next-desolates-map/components/SpaceNavigation/Inhabitants.tsx

```diff
@@ -0,0 +1,90 @@
+import { useContext, useEffect, useState } from "react";
+import { CSSTransition } from "react-transition-group";
+import { SpaceRendererContext } from "../../pages/_app";
+import Loader from "../Loader/Loader";
+
+const PER_PAGE = 3;
+
+export default function Inhabitants() {
+    const { selectedPlanet } = useContext(SpaceRendererContext);
+    const [page, setPage] = useState(1);
+
+    const [nfts, setNFTs] = useState([]);
+    const [maxPages, setMaxPages] = useState<number>(-1);
+
+    const [loading, setLoading] = useState(true);
+
+    async function fetchNFTs() {
+        setLoading(true);
+        const res = await fetch(
+            `/api/allNFTsForOwner/${selectedPlanet?.ownerAddress}?perPage=${PER_PAGE}&page=${page}&includeTotal=1`
+        );
+
+        if (res.status === 200) {
+            const resData = await res.json();
```

Comment:
So this component gets removed from the DOM every time a new planet is selected, meaning it _shouldnt_ allow for the nfts of another planet to be loaded. This relates to the memory leak error [here](https://github.com/emperorgurgeh/desolates-map/pull/10#discussion_r739772242) because this exact situation occurred, but it was unable to update the nfts on the unmounted component.

Reply:
resolved by adding a `mounted` state